release notes/v0.41.0.md

```diff
@@ -0,0 +1,70 @@
+- [#2716](https://github.com/grafana/k6/pull/2716) refactored how custom JS tags are applied to metrics and cleaned up validation for invalid tag values.
+
+We also have a couple of significant improvements that will help us develop exciting new features soon:
+
+### Metric time series ([#2594](https://github.com/grafana/k6/pull/2594))
+
+Previous to [#2594](https://github.com/grafana/k6/pull/2594), k6 didn't have an efficient way to group metric samples with the same tags. It meant that a whole class of applications for processing and aggregating metrics were nearly impossible to do or, at best, very inefficient.
+
+At the cost of some minor [breaking changes](#breaking-changes), we now have a performant internal representation to group metric samples with the same tags at the time of the action that generated them, i.e. the time of metric measurement. With this, k6 can efficiently group samples for the same action (e.g. an HTTP request to a specific URL) over time and construct time series with them.
+
+### Internal support for high-cardinality metrics metadata ([#2726](https://github.com/grafana/k6/pull/2726), [#2727](https://github.com/grafana/k6/pull/2727))
+
+Besides the previously described efficient grouping of metric samples into time series, which are meant for relatively low-cardinality data, k6 needed some way to attach high-cardinality metadata as well. This is necessary for data like unique or random identifiers, e.g. Trace and Span IDs (in distributed tracing), user IDs (e.g. in tests with huge data sets), etc.
+
+This has been added in k6 v0.41.0, and the `vu` and `iter` system tags have been transformed into metadata (see the [breaking changes section](#breaking-changes) above), but it is not yet accessible from user scripts. There is no JavaScript API to modify this metadata, only built-in k6 Go modules and [xk6](https://github.com/grafana/xk6) Go extensions can make use of it, for now.
+
+## Roadmap and future plans
+
+### Prometheus remote-write output
+
+Because big parts of our planned metrics refactoring have been merged, we'll finally be able to refactor and improve the [Prometheus remote-write output extension](https://github.com/grafana/xk6-output-prometheus-remote/). Our goal is to make it more efficient, polish its rough edges, and merge it into the k6 core as a built-in output module in the near future.
+
+### Tracing support via high-cardinality metrics metadata
+
+We now support attaching high-cardinality metadata to metric samples, though only from Go code, for now. We plan to expose this functionality via JavaScript APIs as well, so we will be able to [attach unique trace and span IDs](https://github.com/grafana/k6-jslib-httpx/issues/9) to every metric measurement from JS code, without negatively affecting other parts of k6.
```

Comment:
```suggestion
We now support attaching high-cardinality metadata to metric samples, though only from Go code for now. We plan to expose this functionality via the JavaScript APIs as well, so we will be able to [attach unique trace and span IDs](https://github.com/grafana/k6-jslib-httpx/issues/9) to every metric measurement from JS code without negatively affecting other parts of k6.
```

Reply:
This whole section got dropped, since it's not certain if the JS APIs will make it in k6 v0.42.0 and we didn't want to over-promise :disappointed: